[Chen Hung-Yu] iP

jchilling opened the pull request:
# MyDuke
> "Stay Hungry, Stay Foolish." - Steve Jobs

MyDuke is a personalized chatbot just for you! It's
- text-based
- easy to learn
- ~~FAST~~ **_SUPER FAST_** to use

All you need to do is,
1. download it from [here](https://github.com/jchilling/ip).
2. double-click it.
3. chat with it!

And it is **FREE** 🦈 

Features:
- [x] Managing Tasks
- [x]  Managing Deadlines
- [ ]  Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

````
public static void main(String[] args) {
        new Duke("./data/duke.txt").run();
}

````

Commit:
Add Gradle support

jchilling committed:
Add skeletal version of Duke

jchilling committed:
Add the ability to store and display user's text

jchilling committed:
Add classes and the ability to mark tasks as done

jchilling committed:
Add support for tracking Todos, Deadlines, Events

jchilling committed:
Add skeletal A-TextUiTesting

jchilling committed:
Handle errors and add exceptions

jchilling committed:
Add support for deleting tasks + use of collections & enums

jchilling committed:
Add test cases and documentations

jchilling committed:
Add a save behavior

jchilling committed:
Add parse date and time function

jchilling committed:
Merge branch 'branch-Level-8' & 'branch-Level-7'

jchilling committed:
Modify the code to more OOP

jchilling committed:
Divide classes into packages

jchilling committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

jchilling committed:
Fix minor issues

jchilling committed:
Add JUnit test cases and gradle checkstyle support

jchilling committed:
Add JavaDoc

jchilling committed:
Tweak the code to comply with coding standard

jchilling committed:
Add ability to find tasks by searching a keyword

jchilling committed:
Update README.md

jchilling committed:
Merge branch 'branch-A-CodingStandard'

jchilling committed:
Merge branch 'branch-Level-9'

jchilling committed:
Merge branch 'master' of https://github.com/jchilling/ip

jchilling committed:
Fix minor style issues

jchilling committed:
Add partial implementation of GUI

jchilling committed:
Add GUI

jchilling committed:
Merge branch 'branch-Level-10'

ShawnChew commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,170 @@
+    public String displayDescription() {
+        return description;
+    }
+
+
```
Small issue, but I noticed that line breaks are not consistent in your code. Perhaps you could standardize the line breaks to make the code look cleaner.

ShawnChew commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,71 @@
+     */
+    public ArrayList<Task> load() throws FileNotFoundException {
+        File localStorage = new File(filePath);
+        ArrayList<Task> fromStorage = new ArrayList<>();
```
Since it is a variable, the name should be a noun, and plural in this case. Perhaps you could rename it as tasksFromStorage?

ShawnChew commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,170 @@
+    /**
+     * Show line to the user.
+     */
+    public void showLine() {
+        description = "_______";
+    }
```
Do you need this method?

ShawnChew commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,137 @@
+public class Parser {
+
+    /**
+     * Make sense of the user's input by parsing.
+     * @param userInput The command user input in Duke command line.
+     * @return A Command object representing user command.
+     * @throws DukeException when user command is invalid.
```
There should be an empty line between description and parameter section. I noticed the same issue in several other places too.

ShawnChew commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,75 @@
+    }
+
+    @Override
+    public boolean compareDate(LocalDate date) {
```
Perhaps the name of the method could be isDateEqual to make it more boolean sounding.

ShawnChew reviewed (commented):
Looks good overall! Just take note of the line break in the header comment. Names of variables and methods are generally appropriate. 👍

ChryslineLim commented on src/main/java/duke/commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,43 @@
+public class DeadlineCommand extends Command {
+
+    public static final String COMMAND_WORD = "deadline";
+    private String[] userDescription;
```
Perhaps ```userDescriptions``` may be a better name as there are multiple Strings saved?

ChryslineLim commented on data/duke.txt:
```diff
@@ -0,0 +1 @@
+T | 0 | sleep
```
Are both duke.txt and tasks.txt file being used? If not, you might want to delete one of them.

ChryslineLim commented on src/main/java/duke/commands/FindCommand.java:
```diff
@@ -0,0 +1,34 @@
+package duke.commands;
+import java.util.ArrayList;
```
Maybe you can leave a line after package statement for consistency with your other files.

ChryslineLim commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,256 @@
+            assertEquals("Please enter a date.", e.getMessage());
+        }
+    }
+}
```
I like that you wrote a comprehensive list of tests for your parse function, and they are all named in the right format!

ChryslineLim commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,170 @@
+    public void showGetDate(ArrayList<Task> onDateTasks) {
+        String s = "";
+        for (Task t : onDateTasks) {
+            s = s + t + "\n";
```
Not a coding standard violation, but you can consider using StringBuilder to concatenate Strings!

ChryslineLim reviewed (commented):
Overall, I think you did great! There are barely any coding standard violations and I like the way you organised your files into the packages! :)

jchilling committed:
Remove text-ui-test
After implementation of GUI, text-ui is removed.
text-ui-test is redundant

jchilling committed:
Add Java Assertions
There are important assumptions in various part of the code.

Let's add assertions to check for null inputs that should not occur
in the program flow.

jchilling committed:
Improve code quality

There are a several parts of the code that do not demonstrate
good code quality.

Let's,
* seperate each command's parse from the Parse:parse method to avoid
nested and long methods
* follow SLAP

jchilling committed:
TaskList class: use Streams if applicable

jchilling committed:
Merge pull request #1 from jchilling/branch-A-Assertions

Add Java Assertions

jchilling committed:
Merge branch 'master' into branch-A-CodeQuality

jchilling committed:
Merge branch 'master' into branch-A-Streams

jchilling committed:
Merge pull request #2 from jchilling/branch-A-CodeQuality

Improve code quality

jchilling committed:
Merge branch 'master' into branch-A-Streams

jchilling committed:
Merge pull request #3 from jchilling/branch-A-Streams

TaskList class: use Streams if applicable

jchilling committed:
Provide a way to prioritize items

There is a need to prioritize certain tasks over others.

Let's enable the user to attach priorities HIGH, MEDIUM, and LOW to
Todo, Event, and Deadline tasks.

jchilling committed:
Merge branch 'branch-C-Priority'

jchilling committed:
Fix minor coding standard issues

jchilling committed:
Modify FXML for a friendlier GUI

jchilling committed:
Fix incorrect error message

jchilling committed:
Add UserGuide

jchilling committed:
Fix trialing whitespace

jchilling committed:
Fix Dialog Box

Let text fully expand.
